### DISCUSSION TITLE
Confusing example in Thread Context documentation

### DISCUSSION THREAD

**Discussion by MaxAller** [apache/logging-log4j2#3178] [Q&A]
## Description

On [this page](https://logging.apache.org/log4j/2.x/manual/thread-context.html#init) ([in source](https://github.com/apache/logging-log4j2/blob/2.x/src/site/antora/modules/ROOT/pages/manual/thread-context.adoc#initializing-thread-context)), this example is given:

```
LOGGER.debug("Starting background thread for user");

Map<String, String> mdc = ThreadContext.getImmutableContext(); // (1)
List<String> ndc = ThreadContext.getImmutableStack().asList(); // (1)

executor.submit(() -> {
    try (CloseableThreadContext.Instance ignored = CloseableThreadContext
            .putAll(values) // (2)
            .pushAll(messages)) { // (2)

        LOGGER.debug("Processing for user started");
        // ...

    }
});
```

However...there's no connection between the (1) and (2) items. Is `values` supposed to be `mdc`, and `messages` should be `ndc`?

**Comment by vy** [comment]
@MaxAller, you're right, there is room for improvement in that particular example. (Your report indeed qualifies as a documentation bug. I talked with @ppkarwasz, it was a mistake to convert this into a discussion, but... anyway.) Would you mind submitting a PR improving the associated part of the page, please?

**Reply by MaxAller** [reply]
Can do.

**Comment by MaxAller** [comment]
Fix merged in https://github.com/apache/logging-log4j2/pull/3186
